### PR TITLE
re #249: Allow StealthLink to send larger images

### PR DESCRIPTION
**Pull request by Sunderlandkyl**
- ReleaseDataFlag is turned on for all filters when getting the image.
- Catch the bad_alloc exception that is thrown by VTK if an image is still to large, then display an error message and return failure.

The server will still crash if the image is too large in debug mode, however that may be unavoidable, since abort() is called in vtkGenericDataArray in debug mode.

Works on a 512^3 image using 2 bytes for each voxel in both release and debug.

Relevant issue: https://github.com/PlusToolkit/PlusLib/issues/249

### CODE REVIEW

**Comment by lassoan** [comment]
Nice work!

File: src/PlusDataCollection/StealthLink/vtkPlusStealthLinkTracker.cxx [hunk 1]
```diff
@@ -656,18 +656,29 @@ PlusStatus vtkPlusStealthLinkTracker::GetImage(const std::string& requestedImage
...
+      vtkSmartPointer<vtkImageFlip> flipZFilter = vtkSmartPointer<vtkImageFlip>::New();
+      flipZFilter->SetFilteredAxis(2);   // flip z axis
+      flipZFilter->ReleaseDataFlagOn();
+      flipZFilter->SetInputConnection(flipYFilter->GetOutputPort());
+      flipZFilter->Update();
+      imageData->DeepCopy(flipZFilter->GetOutput());
+    }
+    catch (const std::bad_alloc& e)
+    {
+      LOG_ERROR("Error when downloading image: " << e.what() << ". Image may be too large");
```
Comment (lassoan) [inline src/PlusDataCollection/StealthLink/vtkPlusStealthLinkTracker.cxx]: Maybe add "from StealthLink"

    ...downloading the image from StealthLink: ...

File: src/PlusDataCollection/StealthLink/vtkPlusStealthLinkTracker.cxx [hunk 2]
```diff
@@ -656,18 +656,29 @@ PlusStatus vtkPlusStealthLinkTracker::GetImage(const std::string& requestedImage
     examImageDirectoryToDelete = examImageDirectory;
 
     vtkSmartPointer<vtkDICOMImageReader> reader = vtkSmartPointer<vtkDICOMImageReader>::New();
```
Comment (lassoan) [inline src/PlusDataCollection/StealthLink/vtkPlusStealthLinkTracker.cxx]: Is there a particular reason why the reader instantiation is outside the try/catch block?

Comment (Sunderlandkyl) [reply src/PlusDataCollection/StealthLink/vtkPlusStealthLinkTracker.cxx]: The reader is still required later in the function, so I moved the declaration/instantiation outside the try/catch block.

Would it be better to have the declaration outside the try/catch, and move the instantiation back inside the block?

**Pull request merged**